modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/rebalance/DistributionZoneRebalanceEngineTest.java

```diff
@@ -393,6 +368,72 @@ void staleDataNodesEvent(
         verify(keyValueStorage, timeout(1000).times(2)).invoke(any(), any());
     }
 
+    @Test
+    void replicasTriggersAssignmentsChangingOnNonDefaultZones(
+            @InjectConfiguration
+                    ("mock.tables {"
+                            + "table0 = { zoneId = 1 },"
+                            + "table1 = { zoneId = 1 },"
+                            + "table2 = { zoneId = 2 }}")
+            TablesConfiguration tablesConfiguration
+    ) {
+        assignTableIds(tablesConfiguration);
+
+        when(distributionZoneManager.dataNodes(anyInt())).thenReturn(Set.of("node0"));
+
+        createRebalanceEngine(tablesConfiguration);
+
+        rebalanceEngine.start();
+
+        CompletableFuture<Void> changeFuture = distributionZonesConfiguration.change(zonesChange -> zonesChange.changeDistributionZones(
+                zoneListChange -> zoneListChange.update("zone0", zoneChange -> zoneChange.changeReplicas(2))
+        ));
+        assertThat(changeFuture, willCompleteSuccessfully());
+
+        verify(keyValueStorage, timeout(1000).times(2)).invoke(any(), any());
+    }
+
+    @Test
+    void replicasTriggersAssignmentsChangingOnDefaultZone(
```

Comment:
Did you consider test parametrization for replicasTriggersAssignmentsChangingOnNonDefaultZones  and replicasTriggersAssignmentsChangingOnDefaultZone?

Reply:
I did, but the config is injected via a parameter, which clashes with how `@ParameterizedTest` deals with method parameters